sklearn/impute/_iterative.py

```diff
@@ -581,6 +580,37 @@ def fit_transform(self, X, y=None):
             self.n_iter_ = 0
             return super()._concatenate_indicator(Xt, X_indicator)
 
+        # If min or max value are (scalar) None, convert them to - or + inf
+        self._min_value = -np.inf if self.min_value is None else self.min_value
+        self._max_value = np.inf if self.max_value is None else self.max_value
+        # Convert min and max values to shape=(n_features,) if they are scalars
+        if np.isscalar(self._min_value):
+            self._min_value = np.full(X.shape[1], self._min_value)
+        if np.isscalar(self._max_value):
+            self._max_value = np.full(X.shape[1], self._max_value)
+        # If there were any None values in above arrays, convert them to np.nan
+        self._min_value = as_float_array(self._min_value,
```

Comment:
do not use `as_float_array`. Use `check_array` instead.

Reply:
All these pulled into the `_validate_limit` method as you suggested :)